transaction/std/src/memo_builder/gift_code_cancellation_memo_builder.rs

```diff
@@ -57,53 +73,109 @@ impl MemoBuilder for GiftCodeCancellationMemoBuilder {
+            return Err(NewMemoError::MixedTokenIds);
+        }
         self.wrote_change_memo = true;
-        Ok(GiftCodeCancellationMemo::from(self.gift_code_tx_out_global_index).into())
+        Ok(
+            GiftCodeCancellationMemo::new(self.gift_code_tx_out_global_index, self.fee.value)?
+                .into(),
+        )
     }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::test_utils::build_zero_value_change_memo;
-    use std::convert::TryInto;
+    use crate::test_utils::build_change_memo_with_amount;
 
     #[test]
     fn test_gift_code_cancellation_memo_built_successfully_with_index() {
-        // Set the cancellation index and create memo builder
+        // Set the cancellation index and fee and create the memo builder
         let index = 666;
+        let amount = Amount::new(42, 0.into());
+        let fee = Amount::new(1, 0.into());
         let mut builder = GiftCodeCancellationMemoBuilder::new(index);
+        builder.set_fee(fee).unwrap();
 
         // Build the memo payload and get the data
-        let memo_payload = build_zero_value_change_memo(&mut builder).unwrap();
+        let memo_payload = build_change_memo_with_amount(&mut builder, amount).unwrap();
         let memo_data = memo_payload.get_memo_data();
 
-        // Verify memo data
-        let derived_index = u64::from_le_bytes(memo_data[0..8].try_into().unwrap());
-        assert_eq!(index, derived_index);
+        // Recover memo
+        let memo = GiftCodeCancellationMemo::from(memo_data);
+
+        // Check recovered index is correct
+        assert_eq!(index, memo.cancelled_gift_code_index());
+
+        // Check recovered fee is correct
+        assert_eq!(fee.value, memo.get_fee());
     }
 
     #[test]
     fn test_gift_code_cancellation_memo_fails_for_more_than_one_change_memo() {
-        // Set the cancellation index and create memo builder
+        // Set the cancellation index and fee and create the memo builder
         let index = 666;
+        let amount = Amount::new(42, 0.into());
+        let fee = Amount::new(20, 0.into());
         let mut builder = GiftCodeCancellationMemoBuilder::new(index);
+        builder.set_fee(fee).unwrap();
 
         // Attempt to build two change outputs
-        build_zero_value_change_memo(&mut builder).unwrap();
-        let memo_payload = build_zero_value_change_memo(&mut builder);
+        let memo_payload = build_change_memo_with_amount(&mut builder, amount).unwrap();
+        let memo_payload_2 = build_change_memo_with_amount(&mut builder, amount);
+
+        // Ensure we can recover the index and fee for the first change output
+        let memo = GiftCodeCancellationMemo::from(memo_payload.get_memo_data());
+        assert_eq!(index, memo.cancelled_gift_code_index());
+        assert_eq!(fee.value, memo.get_fee());
 
-        // Assert failure for the second output
+        // Assert failure for the second attempted change output
         assert!(matches!(
```

Comment:
Let's use `assert_matches` (`use assert_matches::assert_matches;`), since this crate already depends on that helper.

Reply:
Will use assert_eq!